service/history/workflow/activity.go

```diff
@@ -169,3 +170,23 @@ func GetPendingActivityInfo(
 
 	return p, nil
 }
+
+func PauseActivityById(mutableState MutableState, activityId string) error {
+	if !mutableState.IsWorkflowExecutionRunning() {
+		return consts.ErrWorkflowCompleted
+	}
+
+	ai, activityFound := mutableState.GetActivityByActivityID(activityId)
+
+	if !activityFound {
+		return consts.ErrActivityNotFound
+	}
+	if ai.Paused {
+		// do nothing
```

Comment:
Still think there's value in using request ID for idempotency and otherwise returning an error.
Worth bringing up for discussion if you disagree.
Not blocking this PR.

Reply:
I do disagree, I think we should raise it with product. Usually it is better to NOT return an error - users can add some alerts/metrics to errors. Anyway, lets raise it with Drew (though - when?)